src/CodeGen_LLVM.cpp

```diff
@@ -554,6 +557,54 @@ std::vector<llvm::Constant*> get_constants(llvm::Type *t, It begin, It end) {
     return ret;
 }
 
+BasicBlock *CodeGen_LLVM::get_destructor_block() {
+    if (!destructor_block) {
+        // Create it if it doesn't exist.
+        BasicBlock *here = builder->GetInsertBlock();
+        destructor_block = BasicBlock::Create(*context, "destructor_block", function);
+        builder->SetInsertPoint(destructor_block);
+        builder->CreateRet(ConstantInt::get(i32, -1));
+        builder->SetInsertPoint(here);
```

Comment:
This has a very frustrating (potential) bug: if the builder is inserting into a BasicBlock before the last instruction (as you do in register_destructor), I think this code will forget that, and set the insert point to the end of the BasicBlock.